clippy_lints/src/write.rs

```diff
@@ -212,8 +212,8 @@ impl EarlyLintPass for Pass {
                 }
             }
         } else if mac.node.path == "write" {
-            if let Some(fmtstr) = check_tts(cx, &mac.node.tts, true).0 {
-                if check_newlines(&fmtstr) {
+            if let (Some(fmtstr), _, is_raw) = check_tts(cx, &mac.node.tts, true) {
+                if !is_raw && check_newlines(&fmtstr) {
```

Comment:
I think we could pass the `is_raw` into `check_newlines` and give that function the ability to detect

```rust
write!(foo, "
");
```

and

```rust
write!(foo, r"
");
```

Reply:
sure! implemented in the latest push.